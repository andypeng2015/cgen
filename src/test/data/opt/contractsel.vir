module foo

function w $foo(w %x) {
@start:
  %c = sgt.w %x, 42_w
  br %c, @default, @chk
@default:
  %y = copy.w 5_w
  jmp @cont
@chk:
  %c = slt.w %x, 0_w
  br %c, @yes, @no
@yes:
  %y = copy.w 3_w
  jmp @cont
@no:
  %y = copy.w %x
  jmp @cont
@cont:
  jmp @cont2
@cont2:
  ret %y
}
