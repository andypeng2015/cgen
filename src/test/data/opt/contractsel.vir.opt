module foo

function w $foo(w %x) {
@2:
  %2 = sgt.w %x, 0x2a_w ; @16
  br %2, @6(0x5_w), @4
@4:
  %3 = slt.w %x, 0x0_w ; @17
  %4 = sel.w %3, 0x3_w, %x ; @18
  jmp @6(%4)
@6(%y.1):
  ret %y.1
}
