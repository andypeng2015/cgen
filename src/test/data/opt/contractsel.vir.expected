module foo

function w $foo(w %x) {
@2:
  %0 = sgt.w %x, 0x2a_w ; @14
  br %0, @6(0x5_w), @4
@4:
  %1 = slt.w %x, 0x0_w ; @15
  %2 = sel.w %1, 0x3_w, %x ; @16
  jmp @6(%2)
@6(%y.1):
  ret %y.1
}
